### PR TITLE
Bert example with Pytorch Lightning

### LINKED ISSUE #334: PyTorch Lightning Example.
Dear Kubeflow team,

A Kubeflow Cluster Environment is being added to PyTorch Lightning: https://github.com/PyTorchLightning/pytorch-lightning/pull/7300.

Would be nice for the Kubeflow team to add a PyTorch Lightning Example using this cluster env.

Best,
T.C

### PR DESCRIPTION
**Pull request by jagadeeshi2i**
Added Bert example with PyTorch Lightning. 

- Fixes   #334

### CODE REVIEW

File: examples/bert/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#FROM pytorch/pytorch:latest
```
Comment (Jeffwan) [inline examples/bert/Dockerfile:1]: remove unused lines?

File: examples/bert/pytorch_job_bert.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      imagePullPolicy: Always
+      restartPolicy: OnFailure
+      template:
+        metadata:
+          annotations:
+            sidecar.istio.io/inject: "false"
+        spec:
+          containers:
+            - name: pytorch
+              image: <username>/pytorch_dist_bert:latest
```
Comment (Jeffwan) [inline examples/bert/pytorch_job_bert.yaml:18]: em. This is ok at this moment.  We should host some example images somewhere in the future.

File: examples/bert/pytorch_job_bert.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      template:
+        metadata:
+          annotations:
+            sidecar.istio.io/inject: "false"
+        spec:
+          containers: 
+            - name: pytorch
+              image: <username>/pytorch_dist_bert:latest
+              # For CPU "--accelerator", "ddp_cpu"
+              args: ["--gpus", "1", "--accelerator", "ddp", "--num_nodes", "2"]
```
Comment (Jeffwan) [inline examples/bert/pytorch_job_bert.yaml:47]: Is it possible to detect all available GPU in the code?

File: examples/bert/pytorch_job_bert.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+              resources:
+                limits:
+                    nvidia.com/gpu: 1 # requesting 1 GPU
+              volumeMounts:
+                - mountPath: /model
+                  name: model-volume
+          volumes:
+            - name: model-volume
+              persistentVolumeClaim:
+                claimName: fileserver-claim
```
Comment (Jeffwan) [inline examples/bert/pytorch_job_bert.yaml:60]: can you add more details on the file storage support here?

File: examples/bert/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+**Note**: PyTorch job doesn’t work in a user namespace by default because of Istio [automatic sidecar injection](https://istio.io/v1.3/docs/setup/additional-setup/sidecar-injection/#automatic-sidecar-injection). In order to get it running, it needs annotation sidecar.istio.io/inject: "false" to disable it for either PyTorch pods or namespace.
+
+**Build Image**
+
+```shell
+docker build -f Dockerfile -t <username>/pytorch-dist-bert:latest .
+docker push <username>/pytorch-dist-bert:latest
+```
+
+**Create the mnist PyTorch job**
```
Comment (lowc1012) [inline examples/bert/README.md:16]: BERT PyTorch job

**Review by gaocegege — COMMENTED** [review]
/lgtm
/assign @johnugeorge

**Comment by johnugeorge** [comment]
Thanks @jagadeeshi2i for this!

/approve

**Pull request merged**